src/request.ts

```diff
@@ -19,7 +19,7 @@ export class FilterRegistry extends Registry<RequestFilter> {
 		}
 		else if (test instanceof RegExp) {
 			entryTest = (response, url, options) => {
-				return test.test(url);
+				return test ? (<RegExp> test).test(url) : null;
```

Comment:
I meant at the top of the function block, so it applies to the whole function scope (and might make it easier to eliminate when we have `const` params), but the code could almost revert back the original form, because `null` has also been eliminated from the type by the `instanceof` guard, right?

```typescript
const inTest = test;
/* ... */
else if (inTest instanceof RegExp) {
  entryTest = (response, url, options) => inTest.test(url);
}
```